Tasks/JenkinsDownloadArtifacts/ArtifactDetails/JenkinsRestClient.ts

```diff
@@ -0,0 +1,145 @@
+                }
+
+                return result;
+            }
+
+            return [];
+        });
+
+        handlebars.registerHelper('containsInArray', function(array, value, options) {
+            if (!!array) {
+                for(let i = 0, len = array.length; i < len; i++) {
+                    tl.debug(`checking ${array[i]} ${value}`);
+                    if (!!array[i] && array[i].indexOf(value) > -1) {
+                        return options.fn(this);
+                    }
+                }
+            }
+
+            return options.inverse(this);
+        });
+
+        handlebars.registerHelper('chopTrailingSlash', function(value, options) {
+            var result: any = value;
+            if (!!value && value.substr(-1) === '/') {
+                result = value.substr(0, value.length - 1)
+            }
+
+            return result;
+        });
+    }
+
+    public DownloadJsonContent(urlPath: string, handlebarSource: string, additionalHandlebarContext: { [key: string]: any }): Q.Promise<any> {
+        let defer = Q.defer<any>();
+
+        const endpoint = tl.getInput("serverEndpoint", true);
+        const endpointUrl = tl.getEndpointUrl(endpoint, false);
+        const jobName = tl.getInput("jobName", true);
+        const username = tl.getEndpointAuthorizationParameter(endpoint, 'username', true);
+        const password = tl.getEndpointAuthorizationParameter(endpoint, 'password', true);
+        const strictSSL: boolean = ('true' !== tl.getEndpointDataParameter(endpoint, 'acceptUntrustedCerts', true));
+
+        let requestUrl: string = `${endpointUrl}/job/${jobName}/${urlPath}`;
+        console.log(tl.loc("DownloadingContentFromJenkinsServer", requestUrl, strictSSL));
+
+        request.get({url: requestUrl, strictSSL: strictSSL}, (err, res, body) => {
+            if (res && body && res.statusCode === 200)  {
+                tl.debug(`Content received from server ${body}`);
+                let jsonResult = JSON.parse(body);
+
+                if (!handlebarSource) {
+                    defer.resolve(jsonResult);
+                }
+                else {
+                    try {
+                        tl.debug(`Applying the handlbar source ${handlebarSource} on the result`);
+                        let template = handlebars.compile(handlebarSource);
+                        if (additionalHandlebarContext) {
+                            for(let key in additionalHandlebarContext) {
+                            tl.debug(`Adding additional context {${key} --> ${additionalHandlebarContext[key]}} to the original context`)
+                                jsonResult[key] = additionalHandlebarContext[key];
+                            };
+                        }
+
+                        var result = template(jsonResult);
+                        defer.resolve(result);
+                    }
+                    catch(err) {
+                        tl.debug(`handlebar failed with an error ${err}`);
```

Comment:
Good to log this to console also using a resource string?

Reply:
fixed.